birdhouse/scripts/get-components-json.include.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+# Obtain a JSON representation of components enabled on this platform.
+#
+# Expected result should be similar to:
+# {
+#  "components": [
+#    "bird-house/birdhouse-deploy:components/monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/canarie-api-full-monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/all-public-access",
+#    "bird-house/birdhouse-deploy:optional-components/wps-healthchecks",
+#    "bird-house/birdhouse-deploy:optional-components/secure-thredds",
+#    "bird-house/birdhouse-deploy:optional-components/testthredds",
+#    "bird-house/birdhouse-deploy:optional-components/test-weaver",
+#    "bird-house/birdhouse-deploy:components/weaver",
+#    "bird-house/birdhouse-deploy:components/cowbird"
+#  ]
+# }
+#
+
+# default value in case of error or missing definitions
+export BIRDHOUSE_DEPLOY_COMPONENTS_JSON='{"components": []}'
+if [ -z "${EXTRA_CONF_DIRS}" ]; then
+  echo "No components in EXTRA_CONF_DIRS. Components JSON list will be empty!"
+  exit 0
+fi
+
+# resolve path considering if sourced or executed, and whether from current dir, pavics-compose include or another dir
+BIRDHOUSE_DEPLOY_COMPONENTS_ROOT=$(dirname -- "$(realpath "$0")")
```

Comment:
I am not sure why you need to discover `BIRDHOUSE_DEPLOY_COMPONENTS_ROOT`.  Since this script is being source by `pavics-compose.sh` its `PWD` will always be the same location of `pavics-compose.sh` since `pavics-compose.sh` needs to enforce that so it can find the `docker-compose.yml`.

So no need to perform any `cd` in the first place.

Reply:
Just to avoid getting this issue if we ever end up calling it instead of sourcing it. This is the actual reason why it didn't filter your absolute paths the first time, while it was working on my side. I was sourcing it directly from another dir to test.

Reply:
Ah ok.  Just curious, in what other scenario would you want to just source this one outside of `pavics-compose.sh`?

In fact, testing it outside of `pavics-compose.sh` is dangerous since you will forget to test it with `pavics-compose.sh` !

Reply:
I tested both to be sure afterward, but I prefer to test it directly than having to go though pavics-compose and quickly breaking out of it. Anyway. they should both work for now and simply remain empty if anything fails.